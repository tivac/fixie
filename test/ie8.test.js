"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie8", () => {
    it("should transform :ie8(...) into an ie8 hack (single selector)", () =>
        simple(`
            :ie8(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie8(...) into an ie8 hack (multiple selectors)", () =>
        simple(`
            :ie8(.fooga .booga) {
                color: red;
            }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie8(&) {
                    color: blue;
                }
            }
        `)
    );
});
