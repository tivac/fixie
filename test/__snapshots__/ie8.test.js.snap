// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie8 should transform :ie8(...) into an ie8 hack (multiple selectors) 1`] = `
"@media \\\\0screen {.fooga .booga {
    color: red;
}
}"
`;

exports[`:ie8 should transform :ie8(...) into an ie8 hack (single selector) 1`] = `
"@media \\\\0screen {.fooga {
    color: red;
}
}"
`;

exports[`:ie8 should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    @media \\\\0screen {
    
    .fooga {
        color: blue;
    }
}"
`;
